hibernate-core/src/test/java/org/hibernate/test/hql/size/SizeTest.java

```diff
@@ -114,6 +150,30 @@ public void testSizeWithNestedPathAndImplicitJoin() {
 		} );
 	}
 
+	@Test
+	public void testSizeWithNestedPathAndImplicitJoinAddFetch() {
+		doInJPA( this::sessionFactory, em -> {
+			List<Student> students = em.createQuery(
+					"select student from Student student left join fetch student.teacher t left join fetch t.skills where size(student.teacher.skills) > 0",
+					Student.class
+			).getResultList();
+			assertEquals( countNumberOfJoins( sqlStatementInterceptor.getSqlQueries().get( 0 ) ), 3 );
+			assertEquals( 1L, students.size() );
+		} );
+	}
+
+	@Test
+	public void testSizeWithNestedPathAndImplicitJoinAddFetchDistinct() {
+		doInJPA( this::sessionFactory, em -> {
+			List<Student> students = em.createQuery(
+					"select distinct student from Student student left join fetch student.teacher t left join fetch t.skills where size(student.teacher.skills) > 0",
+					Student.class
+			).getResultList();
+			assertEquals( countNumberOfJoins( sqlStatementInterceptor.getSqlQueries().get( 0 ) ), 3 );
+			assertEquals( 1L, students.size() );
```

Comment:
The above assertion works for the same reason as `#testSizeWithoutNestedPathAddFetchDistinct`.